angrmanagement/data/instance.py

```diff
@@ -1,43 +1,45 @@
-
 import time
 from threading import Thread
 from queue import Queue
-import traceback
+from typing import List, Optional, Type
+import angr
 
 from .jobs import CFGGenerationJob
 from .object_container import ObjectContainer
 from .sync_ctrl import SyncControl
 from ..logic import GlobalInfo
 from ..logic.threads import gui_thread_schedule_async
 
-
 class Instance:
     def __init__(self, project=None):
         # delayed import
-        from ..ui.views.interaction_view import PlainTextProtocol
+        from ..ui.views.interaction_view import PlainTextProtocol, ProtocolInteractor, SavedInteraction
 
         self.workspace = None
 
         self.jobs = []
         self._jobs_queue = Queue()
-        self.simgrs = ObjectContainer([], name='Global simulation managers list')
-        self.states = ObjectContainer([], name='Global states list')
-        self.patches = ObjectContainer(None, name='Global patches update notifier')
-        self._project_container = ObjectContainer(project, "the current angr project")
+
+        self._project_container = ObjectContainer(project, "The current angr project")
         self._project_container.am_subscribe(self.initialize)
-        self.cfg_container = ObjectContainer(None, "the current CFG")
-        self.cfb_container = ObjectContainer(None, "the current CFBlanket")
-        self.interactions = ObjectContainer([], name='Saved program interactions')
-        self.interaction_protocols = ObjectContainer([PlainTextProtocol], name='Available interaction protocols')
-        self.sync = SyncControl(self)
+        self.extra_containers = {}
+        self._container_defaults = {}
+
+        self.register_container('simgrs', lambda: [], List[angr.SimulationManager], 'Global simulation managers list')
+        self.register_container('states', lambda: [], List[angr.SimState], 'Global states list')
+        self.register_container('patches', lambda: None, None, 'Global patches update notifier') # dummy
+        self.register_container('cfg_container', lambda: None, Optional[angr.analyses.cfg.CFGBase], "The current CFG")
```

Comment:
Took me a minute to figure out the busted import. Need to sync on https://github.com/angr/angr/commit/2a3bd28e69ac13d1a9dd8fc762d2aebbc32cd62f#diff-d186f9f9ffc6122f3256e71152e3f0c3